src/wallet/wallet.cpp

```diff
@@ -5019,9 +5021,9 @@ std::shared_ptr<CWallet> CWallet::Create(interfaces::Chain& chain, interfaces::C
         }
     }
 
-    walletInstance->SetBroadcastTransactions(gArgs.GetBoolArg("-walletbroadcast", DEFAULT_WALLETBROADCAST));
-
     {
+        LOCK(walletInstance->cs_wallet);
```

Comment:
isn't it a backport?

Reply:
To do it properly, I'd need to backport https://github.com/bitcoin/bitcoin/pull/20773, the movement of code to `CWallet::AttachChain()` is what helps restrict the scope of `cs_wallet`, which is mirrored here by bracketing code that would've been moved were it backported.

This PR already has more changes than desired so backporting this proper can be deferred to a future backports PR.

Reply:
Udjin's patch sidesteps the `cs_wallet` inconsistent lock order problem, resolved in latest push!